### PR TITLE
refactor(Transformation): convert multipliers to be aggregators - fixes #217

### LINKED ISSUE #217: Convert `Data/Type/Transformation/` Multipliers to extend `CollectionAggregator`
* FloatMultiplier
* Vector2Multiplier
* Vector3Multiplier

Can all extend `CollectionAggregator` as they are all performing an aggregate function.

### PR DESCRIPTION
**Pull request by thestonefox**
The various Multipler transformations now extend CollectionAggregator
as multiplication is an aggregate function across a collection of
values.

Now it's possible to multiple a range of float/vector2/vector3 values
as they extend the base aggregator.

The prefabs have been updated and the general rule is the dynamic
multiplication value is always index 0 and the multipler is always
set to index 1. This way the event updating the dynamic value never
has to bother setting the current index in the collection.

### CODE REVIEW

File: Scripts/Data/Type/Transformation/CollectionAggregator.cs [hunk 1]
```diff
@@ -98,8 +88,29 @@ protected override TOutput Process(TInput input)
...
+            SetElement(input);
             return ProcessCollection();
         }
+
+        /// <summary>
+        /// Preprocesses the value to be set in <see cref="CurrentIndex"/>
+        /// </summary>
+        /// <param name="value">The value to process.</param>
+        /// <returns>The processed value</returns>
+        protected virtual int PreprocessCurrentIndex(int value)
```
Comment (bddckr) [inline Scripts/Data/Type/Transformation/CollectionAggregator.cs]: It looks like this method can be removed and the property above just uses `ProcessIndex` like everything else.

File: Scripts/Data/Type/Transformation/CollectionAggregator.cs [hunk 2]
```diff
@@ -98,8 +88,29 @@ protected override TOutput Process(TInput input)
...
+        {
+            return ProcessIndex(value);
+        }
+
+        /// <summary>
+        /// Processes the given index into a valid index.
+        /// </summary>
+        /// <param name="index">The value to process.</param>
+        /// <returns>The processed index.</returns>
+        protected virtual int ProcessIndex(int index)
```
Comment (bddckr) [inline Scripts/Data/Type/Transformation/CollectionAggregator.cs]:
```suggestion
        protected virtual int WrapAroundAndClamp(int index)
```
Just a suggestion.

File: Scripts/Data/Type/Transformation/CollectionAggregator.cs (hunk 1 above)
Comment (thestonefox) [reply Scripts/Data/Type/Transformation/CollectionAggregator.cs]: This is in anticipation of the code weaving stuff.

The `PreprocessCurrentIndex` will be the method that's auto woven into the `CurrentIndex` field.

But the actual function being done inside `PreprocessCurrentIndex` is used elsewhere so it's separated out.

**Comment by thestonefox** [comment]
Force pushed the changes

**Pull request merged**